### PR TITLE
Add the contributing guidelines

### PR DESCRIPTION
**Pull request by wixette**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+## Submitting changes
+
+For the main branch, all commits must be submitted via a pull request with at
+least one approving reviews from the core team.
+
+For the developers who have write-access to the repository, please consider the
+following branch naming conventions before submitting a pull request:
+
+ * Release branches: `release_<version>`
+ * Personal working branches: `wip_<your-github-id>` or `wip_<your-github-id>_<sequence_no>`
```
Comment (codingpotato) [inline CONTRIBUTING.md:21]: Do we need add task name in the personal working branch name, for example 'wip_<your-github-id>_<task-name>'?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+Thanks for taking the time to contribute!
+
+## Design decisions
+
+There are lots of trade-offs to made when building a new programming
+environment. We will document every significant design decision in the `design`
+dir. Please create a new issue labeled with `design` if you find anything
+undocumented or propose an update for an existing design.
+
+## Submitting changes
```
Comment (codingpotato) [inline CONTRIBUTING.md:12]: Do we need any rule for the commit message?

File: CONTRIBUTING.md (hunk 1 above)
Comment (wixette) [reply CONTRIBUTING.md:21]: OK. I'd add this format as a new optional naming convention for personal working branches.

File: CONTRIBUTING.md (hunk 2 above)
Comment (wixette) [reply CONTRIBUTING.md:12]: Will add a short description to encourage contributors to follow the guideline at https://chris.beams.io/posts/git-commit/

**Comment by wixette** [comment]
New commits submitted. The messages of the new commits have started to follow the guidelines of https://chris.beams.io/posts/git-commit/

Please take another look.

**Review by codingpotato — APPROVED** [review]

**Pull request merged**